### PR TITLE
Testable multi file upload

### PR DESCRIPTION
**Pull request by StevenLeighton21**
Replaces access to the file upload component with a new, extended component

image

It can be one (optional file) or up to 10 files
image

image

image

Updating the max files to greater than one will show the user how many files they can add

image

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -312,11 +316,16 @@ en:
       max_string_length: 'Maximum answer length...'
       min_string_length: 'Minimum answer length...'
       upload_options: 'Autocomplete options...'
+      multiupload_options: 'Number of files...'
```
Comment (njseeto) [inline config/locales/en.yml:319]:
```suggestion
```
I don't think we use this one

File: app/controllers/pages_controller.rb [hunk 2]
```diff
@@ -98,6 +98,33 @@ def common_params
     }
   end
 
+  def multiupload_files_remaining
+    max_files = @page.components.select { |c| c.type == 'multiupload' }.first['max_files'].to_i
```
Comment (njseeto) [inline app/controllers/pages_controller.rb:102]:
```suggestion
    max_files = @page.components.select(&:multiupload?).first['max_files'].to_i
```

File: app/controllers/pages_controller.rb [hunk 3]
```diff
@@ -98,6 +98,33 @@ def common_params
...
+    answered = false
```
Comment (njseeto) [inline app/controllers/pages_controller.rb:103]: can you help me understand what's happening here? I don't fully understand why we are assigning this variable to a boolean and then using it in the if/else statements

**Review by njseeto — COMMENTED** [review]
Looks good, just a couple of comments and one question.
I can't comment on the validators as I wrote them and the js files look fine to me but might be worth getting Chris to have a look over them.

File: app/controllers/pages_controller.rb (hunk 3 above)
Comment (StevenLeighton21) [reply app/controllers/pages_controller.rb]: It wasn't actually being used, it just needed to exist for the template to work on the edit page. Have removed!

**Review by njseeto — APPROVED** [review]

**Pull request merged**